packages/SwingSet/src/kernel/kernel.js

```diff
@@ -428,18 +428,13 @@ export default function buildKernel(
     // eslint-disable-next-line no-use-before-define
     assert(vatWarehouse.lookup(vatID));
     const vatKeeper = kernelKeeper.provideVatKeeper(vatID);
-    const crankNum = kernelKeeper.getCrankNumber();
-    const deliveryNum = vatKeeper.nextDeliveryNum(); // increments
     const { meterID } = vatKeeper.getOptions();
-    /** @type { SlogFinishDelivery } */
-    const finish = kernelSlog.delivery(vatID, crankNum, deliveryNum, kd, vd);
     // Ensure that the vatSlogger is available before clist translation.
-    kernelSlog.provideVatSlogger(vatID);
+    const vs = kernelSlog.provideVatSlogger(vatID).vatSlog;
     try {
       // eslint-disable-next-line no-use-before-define
-      const deliveryResult = await vatWarehouse.deliverToVat(vatID, vd);
+      const deliveryResult = await vatWarehouse.deliverToVat(vatID, kd, vd, vs);
       insistVatDeliveryResult(deliveryResult);
```

Comment:
We do `insistVatDeliveryResult(deliveryResult)` twice now. Is that on purpose? Belt and suspenders?

Reply:
yeah, pretty much.. the kernel really wants the check because it unpacks and uses the result, but copying it into vat-warehouse.js was mostly just paranoia